### ISSUE TITLE
Silly FR: Dark Mode

### ISSUE THREAD

**Issue by ghost** [hydrusnetwork/hydrus#18]
TLDR: I bought an AMOLED tablet, which I'm going to mainly use for Hydrus remote access via the Splashtop/touch keyboard method I mentioned previously.

If it were possible to set Hydrus to some sort of "night mode" where all the vast white expanses (the backgrounds of taglists and the main view, and especially the background of the Filter view) could be set to black or dark gray, I'd save a lot of battery. (I tried High Contrast mode on the PC, but it doesn't actually affect anything beyond the window borders).

If this would take more than like 15 minutes to code, don't bother - I just wondered if it might be a simple option to add.

**Comment by disposablecat** [comment]
(also, while we're talking about quality of life improvements to the Filter view, an "always zoom smaller images to fit" option would be great - fit works for images larger than the window, but it'd be nice to have smaller images fitted as well. Unless this already exists and I'm missing it.)

**Comment by hydrusnetwork** [comment]
I have added custom colours for next week. This isn't exactly what you want, but I hope it will fill the role. If you find yourself annoyingly going back and forth between different colour modes, I can just add a checkbox for night mode to draw all backgrounds 80% darker or whatever. Let me know once you have played with the new options.

Making small images zoom to fit is a great idea; I'll try to fit it in this week.

**Comment by hydrusnetwork** [comment]
'Fit to canvas' will be in tomorrow, under the new file->options->media options page. I am surprised at how much I like it; I think I'll turn it on for my normal laptop browsing.

**Comment by disposablecat** [comment]
Both the new options work great. Things just look so much better on black, quite aside from the battery savings, and fitting things just feels better - yeah, you have to worry about blowing stuff up and pixelation and so forth, but at the same time you're no longer wasting all that screen space. Thank you!

I'm going to leave it in dark colors all the time - no reason not to, IMO.

A few follow on thoughts:

-The only thing that doesn't appear to be colorable (besides modal dialogs and window frames, which I don't give a fuck about) is the "selection tags" box background/search tag list background. Is that a technical limitation of the method you're using? If not, I'd love to be able to black that out too.

-I've been noticing this, but with fit to canvas it's more pronounced: I have a lot of images with aspect ratios that pretty much match my monitor. These images tend to cut off the tag list on the right hand side of the fullscreen viewer. This is great because I get to use the whole screen for the image, but less great because it means I can't see tags for those without popping up Manage Tags, which I'd rather not do at all if I can avoid it because it blocks filter hotkey input.

So: Is it at all possible to make the tag list "float" over the image? I understand if not, but I thought I'd ask. For the record I'm not thinking "float the bare text" because that causes all sorts of color issues; I'm thinking "cut a section of (maybe 50-75% opacity) background out that covers barely enough area to fit the tag list on top of it, then float that".

**Comment by hydrusnetwork** [comment]
Tags box background colour will be customisable in v146!

I want to make the tags box in the media view a true tags box with right-click support and so on, and have it float pretty much as you describe, along with the file and ratings information, in a near-future version. It is doable, but it'll take a little bit of work. I'll have it auto-hide, like the mouse cursor currently does.
